Sources/Location/LocationManager.swift

```diff
@@ -31,7 +34,21 @@ class LocationManager {
     
     var beaconObservers = ObserverSet<Set<Beacon>>()
     
-    init(context: NSManagedObjectContext, eventQueue: EventQueue) {
+    init(
+        context: NSManagedObjectContext,
+        eventQueue: EventQueue,
+        maxGeofenceRegionsToMonitor: Int,
+        maxBeaconRegionsToMonitor: Int
+    ) {
+        self.maxGeofenceRegionsToMonitor = maxGeofenceRegionsToMonitor
+        self.maxBeaconRegionsToMonitor = maxBeaconRegionsToMonitor
+        let theoreticalMaximumGeofences = 20
+        if maxGeofenceRegionsToMonitor > theoreticalMaximumGeofences {
+            assertionFailure("You may only specify that Rover can monitor up to \(theoreticalMaximumGeofences) geofences at a time.")
```

Comment:
In this case I would say it's more appropriate to use `fatalError` since the developer is _guaranteed_ to execute this code path during development and we can't really recover from this.